actions/global_actions.jsx

```diff
@@ -300,9 +298,55 @@ export async function redirectUserToDefaultTeam() {
     state = getState();
 
     const user = getCurrentUser(state);
+
+    // If the user is a guest, chances are that only the teams are
+    // loaded and, as guests are not necessarily members of the
+    // default channel, we need to find out their next available team
+    // and get channels and memberships for it to be able to route the
+    // user
+    if (Utils.isGuest(user)) {
+        const locale = getCurrentLocale(state);
+        const teamId = LocalStorageStore.getPreviousTeamId(user.id);
+
+        let team = getTeam(state, teamId);
+        const myMember = getMyTeamMember(state, teamId);
+
+        if (!team || !myMember || !myMember.team_id) {
+            team = null;
+            let myTeams = getMyTeams(state);
+
+            if (myTeams.length > 0) {
+                myTeams = filterAndSortTeamsByDisplayName(myTeams, locale);
+                if (myTeams && myTeams[0]) {
+                    team = myTeams[0];
+                }
+            }
+        }
+
+        await dispatch(fetchMyChannelsAndMembers(team.id));
```

Comment:
Might `team` still be `null` in some edge case (either `myTeams.length === 0` or `filterAndSortTeamsByDisplayName` removes all?)